### PR TITLE
[MSHARED-1390] Deprecate InvocationRequest#setGoals

### PR DESCRIPTION
**Pull request by slawekjaranowski**
https://issues.apache.org/jira/browse/MSHARED-1390

### CODE REVIEW

**Review by michael-o — CHANGES_REQUESTED** [review]
Can you move the unrelated changes to a separate PR? It is hard to indentify the actual change.

**Review by olamy — APPROVED** [review]

**Review by CrazyHZM — APPROVED** [review]

File: src/main/java/org/apache/maven/shared/invoker/DefaultInvocationRequest.java [hunk 1]
```diff
@@ -471,6 +472,12 @@ public InvocationRequest addArg(String arg) {
         return this;
     }
 
+    @Override
+    public InvocationRequest addArgs(Collection<String> args) {
+        args.stream().filter(arg -> !StringUtils.isBlank(arg)).forEach(this.args::add);
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/invoker/DefaultInvocationRequest.java:477]: Should we really check for empty arg lines?

```
<args>
<arg> </arg>
</args>
```
??

I mean that is a deliberate input, no?

File: src/main/java/org/apache/maven/shared/invoker/InvocationRequest.java [hunk 2]
```diff
@@ -552,6 +553,16 @@ enum CheckSumPolicy {
      */
     InvocationRequest addArg(String arg);
 
+    /**
+     * Add a raw  arguments list to Maven  cli command at the end of other arguments.
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/invoker/InvocationRequest.java:557]: Between "raw" and "arguments"  should be one space

**Review by michael-o — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/shared/invoker/DefaultInvocationRequest.java (hunk 1 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/invoker/DefaultInvocationRequest.java:477]: in `addArg` we do the same ... 
I'm not sure how it works with empty arg

Comment (michael-o) [reply src/main/java/org/apache/maven/shared/invoker/DefaultInvocationRequest.java:477]: I'd say, shit in, shit out, no?

File: src/main/java/org/apache/maven/shared/invoker/InvocationRequest.java (hunk 2 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/invoker/InvocationRequest.java]: done

File: src/main/java/org/apache/maven/shared/invoker/DefaultInvocationRequest.java (hunk 1 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/invoker/DefaultInvocationRequest.java]: ok, such those are new methods we can simplify it

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #156

**Pull request merged**